### PR TITLE
Make IterableDataset dataloader.__len__ warning clearer

### PR DESCRIPTION
**Pull request by ssnl**
Based on discussion with @jlucier (https://github.com/pytorch/pytorch/pull/38925#issuecomment-655859195) . `batch_size` change isn't made because data loader only has the notion of `batch_sampler`, not batch size. If `batch_size` dependent sharding is needed, users can still access it from their own code.

### CODE REVIEW

**Comment by jlucier** [comment]
Here's more verbose, but concrete example to illustrate the problem described in the edits made in this PR: [#38925 comment](https://github.com/pytorch/pytorch/pull/38925#issuecomment-646688126)

File: torch/utils/data/dataloader.py [hunk 1]
```diff
@@ -116,14 +116,23 @@ class DataLoader(Generic[T_co]):
...
-              When :attr:`dataset` is an :class:`~torch.utils.data.IterableDataset`,
-              ``len(dataset)`` (if implemented) is returned instead, regardless
-              of multi-process loading configurations, because PyTorch trust
-              user :attr:`dataset` code in correctly handling multi-process
-              loading to avoid duplicate data. See `Dataset Types`_ for more
-              details on these two types of datasets and how
-              :class:`~torch.utils.data.IterableDataset` interacts with `Multi-process data loading`_.
+    .. warning:: ``len(dataloader)`` heuristic is based on the length of the sampler used.
+                 When :attr:`dataset` is an :class:`~torch.utils.data.IterableDataset`,
+                 it instead reutrns an estimate based on ``len(dataset) / batch_size``, with proper
```
Comment (zou3519) [inline torch/utils/data/dataloader.py:121]: typo: reutrn -> returns

File: torch/utils/data/dataloader.py [hunk 2]
```diff
@@ -116,14 +116,23 @@ class DataLoader(Generic[T_co]):
...
+                 rounding depending on :attr:`drop_last`, regardless of multi-process loading
+                 configurations. This represents the best guess PyTorch can make because PyTorch
+                 trust user :attr:`dataset` code in correctly handling multi-process
```
Comment (zou3519) [inline torch/utils/data/dataloader.py:124]: "trust user dataset code" -> "trusts user dataset code", because "PyTorch" is singular

**Review by zou3519 — COMMENTED** [review]
lgtm aside from the return typo -- let me know if you want to fix it or if I should just merge this

**Comment by SsnL** [comment]
@zou3519 Typo fixed!

**Review by zou3519 — APPROVED** [review]

**Comment by SsnL** [comment]
@zou3519 Thank you! Can this doc patch still make to 1.6?

**Comment by zou3519** [comment]
It should be able to make it. The internal CI turnaround for this PR will probably be ~3 hours; once this has been merged we can approve and merge the cherry-pick. All of that should be possible by the end of the day today

**Pull request closed without merging**